puzzles/solutions/2022/d02/shapes.py

```diff
@@ -0,0 +1,29 @@
+"""Various constants used for Rock Paper Scissors shapes and other data about them."""
+
+
+OUR_ROCK = "X"
+OUR_PAPER = "Y"
+OUR_SCISSORS = "Z"
+OPPONENT_ROCK = "A"
+OPPONENT_PAPER = "B"
+OPPONENT_SCISSORS = "C"
+
+OUR_TO_OPPONENT_SHAPE_DEFEAT = {
```

Comment:
The different dictionaries here have different logical levels of abstractions.
The first two tell you what is the defeating shape, i.e., it is in the level of the general rock-paper-scissors game.
The third one tells you what input characters are the same shape, i.e., it is in the level of the definition of the problem's input -- not actually related to the game.
The fourth one tells you what the score is for each shape, i.e., it is in the level of the rock-paper-scissors tournament -- one level above the general game. Additional inconsistency is that some scores are defined here as a constant, and others are defined in the code itself.

I suggest you separate the levels of abstraction more clearly:
1. First, parse the problem input into some object describing the rock-paper-scissors game. Note that you don't really need to continue with the "our" and "opponent" shape distinction: If you represent a turn as a pair of shapes (our, opponent), you can use the same object for "our" and for "opponent" rock.
2. Define a function which reports the result of the turn, at the level of the rock-paper-scissors game -- So nothing about scores yet. This function can be a dictionary with all possible turns as keys are the corresponding values representing the result in some way.
3. Define a function which reports the score of the turn, at the level of the rock-paper-scissors tournament, based on the turn pair (from step 1) and the turn's result (from step 2).